File: frontend/src/types/recognition.ts
export interface RecognitionResult {
  success: boolean;
  employee_id?: number;
  employee_code?: string;
  full_name?: string;
  department?: string;
  employee_status?: string;
  similarity_score?: number;
  status?: string; // "match", "borderline", "unknown"
  error?: string;
  liveness_passed: boolean;
  quality_passed: boolean;
}

File: backend/schemas/recognition.py
from pydantic import BaseModel
from typing import Optional

class RecognitionResult(BaseModel):
    success: bool
    employee_id: Optional[int] = None
    employee_code: Optional[str] = None
    full_name: Optional[str] = None
    department: Optional[str] = None
    employee_status: Optional[str] = None
    similarity_score: Optional[float] = None
    status: Optional[str] = None # "match", "borderline", "unknown"
    error: Optional[str] = None
    liveness_passed: bool = False
    quality_passed: bool = False
